coil-gif/src/main/java/coil/transform/AnimatedTransformation.kt

```diff
@@ -0,0 +1,42 @@
+package coil.transform
+
+import android.graphics.Canvas
+import coil.size.Size
+import android.graphics.PixelFormat as AndroidPixelFormat
+
+/**
+ * An interface for applying transformation on GIFs, animated WebPs, and animated HEIFs.
+ */
+interface AnimatedTransformation {
+
+    /**
+     * Apply transformation on [canvas]
+     *
+     * Note: Do not allocate objects within if you are using [coil.decode.GifDecoder] as this method would be invoked
```

Comment:
I would keep it simple with `Do not allocate objects in this method as it will be invoked for each frame of the animation.`